roms_tools/utils.py

```diff
@@ -0,0 +1,125 @@
+from numbers import Integral
+
+import numpy as np
+import xarray as xr
+
+
+def partition(ds: xr.Dataset, nx: int = 1, ny: int = 1) -> list[xr.Dataset]:
+    """
+    Split a ROMS dataset up into nx by ny spatial tiles.
+    """
```

Comment:
We should expose a partitioning tool in the public API (as part of a new `utils` module perhaps), but I'm not sure if it should operate on files directly or on xarray Datasets (like this function does).

Reply:
I think it's great to have it (more generally) work on xarray Datasets. Then we can invoke it as part of the `.save` method. 

But we should definitiely add a page to the documentation that illustrates how you would use `partition` on already saved files, via
```
xr.open_dataset()
partition()
xr.save_mfdataset()
```